drivers/staging/vc04_services/interface/vchiq_arm/vchiq_core.c

```diff
@@ -2407,6 +2407,11 @@ vchiq_init_state(VCHIQ_STATE_T *state, VCHIQ_SLOT_ZERO_T *slot_zero,
 		return VCHIQ_ERROR;
 	}
 
```

Comment:
I can't agree at this point. The id is a static if vchiq_init_state is called twice during a probe defer or something else. So the right fix would be to use a fixed 0 and an overflow is really impossible.